loopd/daemon.go

```diff
@@ -161,75 +179,184 @@ func daemon(config *config, lisCfg *listenerCfg) error {
 		log.Infof("REST proxy disabled")
 	}
 
-	mainCtx, cancel := context.WithCancel(context.Background())
-	var wg sync.WaitGroup
+	// Start the main event handlers.
+	d.startEventHandlers()
 
-	// Start the swap client itself.
-	wg.Add(1)
+	// Start the grpc server.
+	d.wg.Add(1)
 	go func() {
-		defer wg.Done()
+		defer d.wg.Done()
 
-		log.Infof("Starting swap client")
-		err := swapClient.Run(mainCtx, server.statusChan)
+		log.Infof("RPC server listening on %s", d.grpcListener.Addr())
+		err = d.grpcServer.Serve(d.grpcListener)
 		if err != nil {
 			log.Error(err)
```

Comment:
What happens to the rest server in this case?

Reply:
`Serve` blocks until the server is shut down. So the error will only be returned if we're already in shutdown anyway. The REST server will also be shut down then.

Reply:
I was thinking about other errors than the shutdown error